providers/context_value_provider.go

```diff
@@ -0,0 +1,21 @@
+package providers
+
+import "context"
+
+const userIdKey contextKey = "userId"
+
+type contextKey string
+
+type ContextValueProvider struct{}
+
+func NewContextValueProvider() *ContextValueProvider {
```

Comment:
not sure why you called it `ContextValueProvider`... what context does it provide? From code I see that it provides current user/session information/authorization information , but not context.